src/Time/Time.php

```diff
@@ -21,6 +21,63 @@
 
 class Time implements TimeContract
 {
+
+    /**
+     * Date format.
+     *
+     * @since 3.0.0
+     *
+     * @var string
+    */
+    private $dateFormat;
+
+    /**
+     * Set the time format.
+     *
+     * @param mixe $format Valiud Date/Time format.
+     *
+     * @since 3.0.0
+     *
+     * @return object
+     */
+     public function setDateFormat($format)
+     {
+        if ($this->validateDateFormat($fornat)) {
+            $this->dateFormat = $format;
+
+            return $this;
+        }
+
+        throw new \InvalidArgumentException("The format {$format} is not valid format", 500);
+        
+     }
+
+    /**
+     * Get the time format.
+     *
+     * @since 3.0.0
+     *
+     * @return string
+     */
+     public function getDateFormat()
+     {
+        return $this->dateFormat;        
+     }
+
+    /**
+     * Validate the time format.
+     *
+     * @param mixe $format Valiud Date/Time format.
+     *
+     * @since 3.0.0
+     *
+     * @return bool
+     */
+     public function validateDateFormat($format)
```

Comment:
Here is the [reference ](https://www.php.net/manual/en/function.checkdate.php#113205)about validating date format approach.

Reply:
oh, we need to validate Date with format or just format

is that what you propose

> Add the validateDateFormat method when setting setDateFormat

Reply:
I think we can validate date format, not just for the date format string.